### DISCUSSION TITLE
How to handle recalculating Budgets after adding past transactions

### DISCUSSION THREAD

**Discussion by AmineI** [firefly-iii/firefly-iii#8507] [Firefly III] [answered]
Hello ! I've been learning to use FireflyIII for a few months now - thanks for your work on this powerful tool.

I still can't fully grasp how to master budgets with auto-correction for unspent & overspent money, and I'd like your knowledge/input on this.
Here's a simplified scenario : 

- Have a 500€ budget for groceries. Auto-grow & reduce budgets is enabled.
- Let's assume you didn't import any transaction in that budget for January.
- On February 1st, the cron job fires, and calculates the new budget for February. 
- Firefly now says you have 1000€ for groceries in February.

- Now, on February 4th, **you add/import a transaction made in January** : You bought 400€ worth of groceries on January 27th. That means you actually had 100€ left for groceries in January.
- Your February budget for groceries is really 600€ - but Firefly does not update that, and still displays 1000€ available in the February budget.

What would be the best way to recalculate/update these budgets so that they are accurate in Firefly in this case ?

This is a simplified example, which unfortunately would occur every month for me : many of my transactions are directly debited from my account with some sort of autopay. I try to import/reconcile FireflyIII regulary. I believe doing that every last day of the month wouldn't be a reliable or efficient solution, because my transactions can take days to show up in my bank account history.

I looked into forcing the cron-job after such past reconciliations, but it looks like it would duplicate all budgets and cause issues.. 
Should this recalculation be done by FireflyIII automatically instead ?

Thanks again for your time and guidance in these discussion spaces !

**Comment by JC5** [comment, marked as answer]
https://docs.firefly-iii.org/how-to/firefly-iii/advanced/cron/#extra-information

Try a cron job like this:

```
0 3 7 * * /usr/bin/php /var/www/html/artisan firefly-iii:cron --force --date=$(date "+\%Y-\%m-")01
```

This will run on the 7th of the month, but pretends it's the 1st of the month. That gives you a week to get all transactions in place.

Of course, you will not have budget information for the first 7 days.

**Reply by AmineI** [reply]
I did notice this point in the doc - but didn't go with this due to the API documentation of "--force".

So, to make that work correctly, I will disable my current daily cron and replace it with this monthly cron. I understand they shouldn't both run for the first day of the month, to avoid duplicates.

To avoid other issues, I would also get rid of weekly budgets & recurring transactions so that everything is updated by this monthly job, or tweak my daily cron job to not trigger the first day of the month.
I think I'm not missing anything ?

Thanks again for everything - I'll be sending some coffees your way ;) !

**Reply by JC5** [reply]
No that looks good. It's a little more complex for the recurring budgets, but otherwise this should work as expected. You can even run the cronjob twice on the 7th day of the month. Once pretending it's the 1st of then month, and the second time for actually the 7th.
